packages/google/src/google-generative-ai-embedding-model.test.ts

```diff
@@ -26,14 +37,22 @@ describe('GoogleGenerativeAIEmbeddingModel', () => {
     headers?: Record<string, string>;
   } = {}) {
     server.urls[
-      'https://generativelanguage.googleapis.com/v1beta/models/text-embedding-004:batchEmbedContents'
+      BATCH_URL
     ].response = {
       type: 'json-value',
       headers,
       body: {
         embeddings: embeddings.map(embedding => ({ values: embedding })),
       },
     };
+
+    server.urls[SINGLE_URL].response = {
+      type: 'json-value',
+      headers,
+      body: {
+        embedding: { values: embeddings[0] },
+      },
+    }
```

Comment:
ideally do not set it up here. set it up in your test case in lieu of preparejsonresponse since it is for your 2 tests only - otherwise you will not test that the correct route is called